saleor/graphql/meta/mutations/base.py

```diff
@@ -165,10 +166,40 @@ def mutate(cls, root, info: ResolveInfo, **data):
         if not cls.check_permissions(info.context, permissions):
             raise PermissionDenied(permissions=permissions)
 
+        try:
+            instance = cls.get_instance(info, id=data["id"])
+        except ValidationError as e:
+            return cls.handle_errors(e)
+
+        if instance:
+            metadata_instance = get_valid_metadata_instance(instance)
+            old_metadata = dict(metadata_instance.metadata)
+            old_private_metadata = dict(metadata_instance.private_metadata)
+        else:
+            old_metadata = {}
+            old_private_metadata = {}
+
         try:
             result = super().mutate(root, info, **data)
-            if not result.errors:
+
+            instance = result.item
+            if isinstance(instance, ChannelContext):
+                instance = instance.node
+
+            if instance:
+                metadata_instance = get_valid_metadata_instance(instance)
+                new_metadata = metadata_instance.metadata
+                new_private_metadata = metadata_instance.private_metadata
+                has_changed = (
+                    old_metadata != new_metadata
+                    or old_private_metadata != new_private_metadata
+                )
+            else:
+                has_changed = False
```

Comment:
Nitpick: I find it's cleaner to assign a default before performing the logic so e.g. here we'd have:
```python
            has_changed = False
            
            instance = result.item
            if isinstance(instance, ChannelContext):
                instance = instance.node

            if instance:
                metadata_instance = get_valid_metadata_instance(instance)
                new_metadata = metadata_instance.metadata
                new_private_metadata = metadata_instance.private_metadata
                has_changed = (
                    old_metadata != new_metadata
                    or old_private_metadata != new_private_metadata
                )
                
```

Reply:
I agree; it's more readable this way.